File: redis/2022/source/17.php
<?php
$redis = new \Redis();
$redis->pconnect('/tmp/redis.sock');
$redis->flushDB();

$t1 = microtime(true);

for ($i = 0; $i < 100000; $i++) {
    $redis->set("info:" . $i, "val");
}

// $pipe = $redis->pipeline();
// for ($i = 0; $i < 100000; $i++) {
//     $pipe->set("info:" . $i, "val");
// }
// $pipe->exec();

$t2 = microtime(true);
echo $t2 - $t1;

File: redis/2022/source/17.go
package main

import (
	"fmt"
	"strconv"
	"time"

	"github.com/go-redis/redis"
)

func main17() {
	rdb := redis.NewClient(&redis.Options{
		Addr:     "unix:/tmp/redis.sock",
		Password: "",
		DB:       0,
	})

	rdb.FlushDB()

	t1 := time.Now()

	for i := 1; i < 100000; i++ {
		rdb.Set("info:"+strconv.Itoa(i), "val", -1)
	}

	// pipe := rdb.Pipeline()
	// for i := 1; i < 100000; i++ {
	// 	pipe.Set("info:"+strconv.Itoa(i), "val", -1)
	// }
	// pipe.Exec()

	t2 := time.Now()

	fmt.Println(t2.Sub(t1))
}
